plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go

```diff
@@ -210,7 +210,6 @@ func buildControllerRoles() ([]rbacv1.ClusterRole, []rbacv1.ClusterRoleBinding)
 				rbacv1helpers.NewRule("get", "list", "watch").Groups(legacyGroup).Resources("pods").RuleOrDie(),
 				rbacv1helpers.NewRule("update").Groups(legacyGroup).Resources("pods/finalizers").RuleOrDie(),
 				rbacv1helpers.NewRule("get", "list", "watch", "create", "delete").Groups(resourceGroup).Resources("resourceclaims").RuleOrDie(),
-				rbacv1helpers.NewRule("get", "list", "watch", "create", "update", "patch").Groups(resourceGroup).Resources("podschedulingcontexts").RuleOrDie(),
```

Comment:
Here and below, it would be reasonable to add one more ACTION REQUIRED to either drop this role `system:controller:resource-claim-controller` and let it be re-created or alternatively to limit its scope.

Reply:
So if it got created in 1.31, it won't get updated automatically in 1.32? That would be surprising. I've made changes to these policies in the past under the assumption that whatever is specified here will be active when the cluster comes up.

@aramase: you [said](https://github.com/kubernetes/kubernetes/pull/128003#issuecomment-2411698045) that SIG Auth looked at this and that it is okay. Can you confirm?

Reply:
Let's try it...

I brought up a 1.31.1 cluster with local-up-cluster.sh. Then I brought up this branch with `PRESERVE_ETCD=true`. A ResourceClaim that I had created in the old cluster was still there, so the "upgrade" was successful. @soltysh is right, the `system:controller:resource-claim-controller` ClusterRole is still the old one with an entry for `podschedulingcontexts`.

Interesting, this is the first time that someone has mentioned this :hot_face: 

Does this only occur when *removing* something and does *adding* perhaps work automatically?

I added some `ACTION REQUIRED` for this. Editing the roles is probably the easiest path.

Reply:
> Does this only occur when removing something and does adding perhaps work automatically?

Yes, it does. I tried the "1.32 with wiped etcd" to "1.31 with kept etcd" direction and the podschedulingcontexts rules got added.

Interesting, I learned something today! Thanks.

Reply:
@liggitt: okay to proceed like this with removal of rules (no code changes, just some `ACTION REQUIRED`)? If yes, then please unhold. This is the last open.

Reply:
All role / binding reconciliation on upgrade is additive, an upgrade *never* automatically removes permissions or bindings. See https://kubernetes.io/docs/reference/access-authn-authz/rbac/#auto-reconciliation

> At each start-up, the API server updates default cluster roles with any missing permissions, and updates default cluster role bindings with any missing subjects. This allows the cluster to repair accidental modifications, and helps to keep roles and role bindings up-to-date as permissions and subjects change in new Kubernetes releases.

What is in this file (and enabled by feature gates) is ensured to be the *minimum* present roles/permissions/bindings.

Since this was never enabled by default, the action required is *only* for people who enabled the alpha feature. Be sure any action-required notes make that clear.